In [ ]:
from transformers import AutoTokenizer

checkpoint = "distilbert-base-uncased-finetuned-sst-2-english"
tokenizer = AutoTokenizer.from_pretrained(checkpoint)

raw_inputs = [
    "I've been waiting for a HuggingFace Course for a very long time.",
    "I love and hate this so much."
]
inputs = tokenizer(
    raw_inputs,
    padding=True,
    truncation=True,
    return_tensors="pt"
)

In [2]:
from transformers import AutoModelForSequenceClassification

checkpoint = "distilbert-base-uncased-finetuned-sst-2-english"
model = AutoModelForSequenceClassification.from_pretrained(checkpoint)
outputs = model(**inputs)
print(outputs.logits)

tensor([[ 2.0498, -1.7484],
        [-1.8256,  1.8829]], grad_fn=<AddmmBackward0>)


In [3]:
import torch

predictions = torch.nn.functional.softmax(
    outputs.logits,
    dim=1
)

print(predictions)

tensor([[0.9781, 0.0219],
        [0.0239, 0.9761]], grad_fn=<SoftmaxBackward0>)


Finally, we give labels to both sentences using `id2label` field.

In [4]:
print(model.config.id2label)

{0: 'NEGATIVE', 1: 'POSITIVE'}


We can see that; </br> 
1st sentence is labeled as **NEGATIVE**. {NEGATIVE: ~97.81%, POSITIVE: ~2.19%} </br>
2nd sentence is labeled as **POSITIVE**. {NEGATIVE: ~2.39%, POSITIVE: ~97.61%}
